src/Symfony/Component/Security/Core/User/LdapUserProvider.php

```diff
@@ -32,15 +33,15 @@ class LdapUserProvider implements UserProviderInterface
     private $defaultSearch;
 
     /**
-     * @param LdapClientInterface $ldap
-     * @param string              $baseDn
-     * @param string              $searchDn
-     * @param string              $searchPassword
-     * @param array               $defaultRoles
-     * @param string              $uidKey
-     * @param string              $filter
+     * @param LdapInterface $ldap
+     * @param string        $baseDn
+     * @param string        $searchDn
+     * @param string        $searchPassword
+     * @param array         $defaultRoles
+     * @param string        $uidKey
+     * @param string        $filter
      */
-    public function __construct(LdapClientInterface $ldap, $baseDn, $searchDn = null, $searchPassword = null, array $defaultRoles = array(), $uidKey = 'sAMAccountName', $filter = '({uid_key}={username})')
+    public function __construct(LdapInterface $ldap, $baseDn, $searchDn = null, $searchPassword = null, array $defaultRoles = array(), $uidKey = 'sAMAccountName', $filter = '({uid_key}={username})')
```

Comment:
True. And this one is a bit more difficult to fix.

The only solution I see would be to move back the `find()` method in the `Ldap` class, and make the `LdapClient` class call the method from the `Ldap` class (as it acts as a proxy).

This also means that the `Ldap` class will embed some legacy logic, which I'm not really in favor of, but we don't have much choice, do we?